[AL] Header and Footer

dpswh-dev opened the pull request:
# Description

Created the following layout components according to the requirements:
- Navbar.tsx
- Footer.tsx

Fixes #4 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.

## Screenshot of changes (if appropriate)

dpswh-dev committed:
feat(navbar): initial add Navbar component

dpswh-dev committed:
feat(navbar): add initial navbar

dpswh-dev committed:
feat(navbar): add navlinks and CISCO logo

dpswh-dev committed:
chore: squash commit

dpswh-dev committed:
feat(footer): add footer content

dpswh-dev committed:
chore(branding): merge both navbar and footer into one branch

dpswh-dev committed:
chore: readd existing h1 to prevent merge conflict

dpswh-dev committed:
refactor(branding): remove subtext from CISCO logo

dpswh-dev committed:
refactor: copy pasted App.tsx from repo to prevent merge conflict

baristabarita reviewed (commented):
So far so good! LGTM!

0-BSCode commented:
@dpswh-dev could you leave some screenshots so we can see what it looks like rn? Also, pull from main then push here so the branch is up-to-date with main:

```bash
# From your local branch
git pull origin main
git push
```

Luzefiru commented on src/components/Navbar.module.css:
```diff
@@ -0,0 +1,10 @@
+.navbarMain {
```
Is it possible to define this in the `index.css` file as a [TailwindCSS layer](https://tailwindcss.com/docs/adding-custom-styles#using-css-and-layer) instead? You can place it under the `@layer components { ... }` so that we can stick to TailwindCSS as our CSS single source of truth.

dpswh-dev replied:
Resolved.

Luzefiru commented on src/components/Footer.tsx:
```diff
@@ -0,0 +1,24 @@
+    <footer className="text-xl bg-[#1752F0] p-6 space-y-3">
+      <img
+        className="w-auto h-6 mx-auto"
+        src="src/assets/cisco-logo-white.png"
```
Is there a reason why assets aren't in the `/public` directory? AFAIK, common conventions place static asset files (images, fonts, etc.) in the `public` directory.

dpswh-dev replied:
Oh, right. Resolved. Thank you for this!

Luzefiru commented on index.html:
```diff
@@ -5,6 +5,12 @@
+    <link rel="preconnect" href="https://fonts.googleapis.com" />
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
+    <link
+      href="https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,100..900;1,100..900&display=swap"
```
Let's stick with CISCO branding and use [Poppins](https://fonts.google.com/specimen/Poppins) as our body copy font.

dpswh-dev replied:
Alright!

Luzefiru reviewed (changes requested):
Some comments need to be addressed.

dpswh-dev committed:
refactor(branding): resolve remaining issues

dpswh-dev committed:
refactor(branding): remove the useless class 'font-roboto' from navbar and footer

Resolved: 1) Font, 2) Moved assets to public/, and 3) Removed Navbar.module.css

dpswh-dev committed:
feat(footer): footer stays at the bottom regardless of content size

Prevented overscrolling on y-axis through 'overscroll-y-none' Tailwind class.

dpswh-dev committed:
refactor(): update img source path to root path

dpswh-dev committed:
feat(layout): combine navbar and footer into a single Layout component

For reusability purposes across different routes.

dpswh-dev committed:
refactor(layout): change encapsulating tag to react fragment

dpswh-dev commented:
@Luzefiru 

Kindly please check the updated PR. I have worked on the following as requested:
- [x] Make Header and Footer to a Layout component for reusability across routes
- [x] Make Footer float down to the bottom regardless of body content 
- [x] Desktop Responsiveness
- [x] Set default font to Poppins in `index.css` 

For a quick demo, please refer to the video.

https://github.com/user-attachments/assets/80fe4206-37f8-42ca-a97d-f11c59bb63f7

# Layout.tsx
![A screenshot of the Layout.tsx component](https://github.com/user-attachments/assets/fb5fb650-d519-48ae-bf2a-be0237bda256)

> [!NOTE]
> I haven't tested the Layout component across different routes with something like React Router, which I presume will be our goto choice for Routing and Navigation, since it hasn't been installed yet. Will it be alright to take the liberty to test it on my machine by installing React Router? Thank you.

dpswh-dev committed:
feat(branding): set default font-family to Poppins in index.css

Luzefiru commented:
> For a quick demo, please refer to the video.

Looks amazing. Great work!


> Note
> 
> I haven't tested the Layout component across different routes with something like React Router, which I presume will be our goto choice for Routing and Navigation, since it hasn't been installed yet. Will it be alright to take the liberty to test it on my machine by installing React Router? Thank you.

Yeah, feel free to install React Router in this PR. It'll be out choice for routing since we are using Vite + React.

Thank you, @dpswh-dev!

dpswh-dev committed:
feat(router): add pages/ to store different pages of the app

dpswh-dev committed:
feat(router): add proper navbar and footer Layout with react router

dpswh-dev committed:
refactor(branding): update focus:ring of navlinks

dpswh-dev committed:
feat(branding): extend tailwind.config.js to include colors from Figma design

Applied and replaced the Navbar and Footer colors to use extended colors.

dpswh-dev committed:
Merge branch 'main' into feat/branding

Luzefiru reviewed (approved):
LGTM!
